packages/prelude/src/function.js

```diff
@@ -3,6 +3,9 @@
 // id :: a -> a
 export const id = x => x
 
+// pipe :: (a -> b) -> (b -> c) -> (a -> c)
+export const pipe = (f, g) => x => g(f(x))
+
 // compose :: (b -> c) -> (a -> b) -> (a -> c)
 export const compose = (f, g) => x => f(g(x))
```

Comment:
Here’s another variant of `compose`, which handles more than just two functions:
```javascript
const compose = (...fns) => fns.reduce((f, g) => (...args) => f(g(...args)))

// Usage : compose functions right to left
// compose(minus8, add10, multiply10)(4) === 42
//
// The resulting function can accept as many arguments as the first function does
// compose(add2, multiply)(4, 10) === 42
```

What do you think, @briancavalier?

Reply:
@Frikki, that can work too, there may be a few caveats that comes to mind

- i think you will lose performance because of v8's deterministic inlining of functions
- current versions of typescript & flow(i assume) are bad at inferring variadic functions(`typescript`, possibly `flow`)

If you don't mind losing performance and/or type validation at runtime(flow), IDE(typescript vscode) and transpiration variadic approach is better then current compose/pipe


btw, variadic `compose` and `pipe` can be reused using the current implementation of pipe and compose (:
````typescript
const cx = (...fns) => fns.reduceRight(compose)
const px = (...fns) => fns.reduce(pipe)
````

Reply:
@nissoh, I think you’re right about the performance loss, unless V8 has become extremely clever.

The current implementation cannot be used the way you describe. Try yourself:

```
const compose = (f, g) => x => f(g(x))

const cx = (...fns) => fns.reduceRight(compose)

const add2 = x => x + 2

const multiply = (x, y) => x * y

cx(add2, multiply)(4, 10) // NaN
```

Reply:
@Frikki yeah, to solve that we have to curry multiply

````typescript
const compose = (f, g) => x => f(g(x))

const cx = (...fns) => fns.reduceRight(compose)
const add2 = x => x + 2
const multiply = curry2((x, y) => x * y)
cx(add2, multiply)(4)(10) // 60
````